packages/frontend/src/screens/app/screens/tournaments/screens/tournament/screens/match/match-view.tsx

```diff
@@ -14,24 +16,54 @@ export interface MatchViewMatch {
 	secondTeam: Team;
 	winner: Team | null;
 	contested: boolean;
+	tournament: { id: string; owner: { id: string } };
+	finalized: boolean;
 }
 
 interface Props {
 	reportWin: (teamId: string) => void;
 	reportContested: (contested: boolean) => void;
+	reportFinalized: () => void;
 	match: MatchViewMatch;
 	loading: boolean;
+	tournament: {
+		id: string;
+		owner: { id: string };
+	};
 }
 
-const MatchView: React.FC<Props> = ({ reportWin, reportContested, match, loading }) => {
+const MatchView: React.FC<Props> = ({
+	reportWin,
+	reportContested,
+	reportFinalized,
+	match,
+	loading,
+	tournament,
+}) => {
+	function firstTeamRetcon() {
+		reportWin(match.firstTeam.id);
+		console.log(reportFinalized());
+	}
+	function secondTeamRetcon() {
+		reportWin(match.secondTeam.id);
+		console.log(reportFinalized());
```

Comment:
reportFinalized is important, even though it is in a console.log + 2 mutations should not be needed to complete an action

Reply:
It shouldn't be in a console.log, we used used it for debug and forgot. But I dont agree on the 2 mutations part. We will need an independent finalize in other places and its kinda wonk imo to add a lot of mutations for actions that are 90% the same if we would combine these into one.